### DISCUSSION TITLE
Save Data from Non-Visible Fields

### DISCUSSION THREAD

**Discussion by moskoweb** [filamentphp/filament#5677] [General]
Hello,

I have a form with Toggle, which if active appears more fields to fill in, however, there may be situations of the user disabling Toggle, and the fields disappear, and I noticed that when they disappear they are also no longer saved, that is, I lose the information of the fields, if Toggle is disabled, and the fields are without visibility.

Is there a way to prevent the data from being lost, or to submit this information, even if it is no longer visible to the user?

**Comment by pxlrbt** [comment]
What's the use case of saving that data when it's hidden? If you need to provide defaults you can use the mutation methods: https://filamentphp.com/docs/2.x/admin/resources/creating-records#customizing-data-before-saving

**Reply by moskoweb** [reply]
The situation of temporary deactivation of the resource by the customer, when he needs to make some internal change, in addition to the fact that even with the feature disabled, the information of the non-visible fields is used in other activities that the customer may want to do.

It would not be defining standard values, but keeping the values that the client has already entered, the idea would be to hide the field so as not to make the form large, but to keep the values.

**Reply by pxlrbt** [reply]
So we are talking about Edit forms. Is the value overwritten, if the field is `->hidden()`? I don't think that should be the case and it should only be updated if the field is present.

**Reply by moskoweb** [reply]
In this case it is a field that contains information, which can be activated or deactivated, only being deactivated, he would not want to lose the information already saved, so that when it is activated again, the user does not need to have to fill in again.


Example, if the status is active, I ask the person to fill in the base_name field, but if they disable it, even though it is no longer visible, I still want the value they have filled in the base_name field to remain saved:

```php
Components\Toggle::make('status')
	->inline(false)
	->reactive(),

Components\TextInput::make('base_name')
	->visible(fn (\Closure $get) => $get('status') === true),
```

**Reply by pxlrbt** [reply]
Use the mutation methods to prefill the current records data and merge the new data into this.
